### PR TITLE
feat: Add creative flight mode

### LINKED ISSUE #2854: [FEATURE] Creative-like flight
### LiquidBounce Branch

Nextgen

### Describe your feature request.

add another type of fly that replicates the fly you get as if you were playing creative, with the glide etc.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Ell1ott**
Adds a new flight mode called "Creative", that works and looks very simular to normal creative flight. It does bypass vanilla flight checks.
Closes #2854

### CODE REVIEW

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/movement/fly/modes/FlyGeneric.kt [hunk 1]
```diff
@@ -69,6 +71,71 @@ internal object FlyVanilla : Choice("Vanilla") {
...
+        player.abilities.allowFlying = true;
+    }
+
+    private fun shouldFlyDown(): Boolean {
+        if (!bypassVanillaCheck) return false
+        if (player.age % 40 != 0) return false
+
+        // check if the player is above a block or in mid-air
+        // if the player is right above a block, we don't need to fly down
+        if (world.getStatesInBox(player.boundingBox.offset(0.0, -0.55, 0.0)).anyMatch { !it.isAir }) return false
```
Comment (superblaubeere27) [inline src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/movement/fly/modes/FlyGeneric.kt:124]: Commentary +1

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/movement/fly/modes/FlyGeneric.kt [hunk 2]
```diff
@@ -69,6 +71,71 @@ internal object FlyVanilla : Choice("Vanilla") {
...
+        return true
+    }
+
+    val repeatable = repeatable {
+        player.abilities.flySpeed =
+            if (mc.options.sprintKey.isPressed && SprintSpeed.enabled) SprintSpeed.speed else speed
+
+        if (forceFlight) player.abilities.flying = true
+
+        if (player.velocity.lengthSquared() > maxVelocity * maxVelocity * maxVelocity) {
```
Comment (superblaubeere27) [inline src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/movement/fly/modes/FlyGeneric.kt]: This line says:
${v_\text {curr}}^2 > {v_\text {max}}^3$

Is this intended? Or did you mean ${v_\text {curr}}^2 > {v_\text {max}}^2$?

**Review by superblaubeere27 — CHANGES_REQUESTED** [review]
Really good code style. Readable and easy to understand 👍🏻 

I think I have spotted a logic error though

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/movement/fly/modes/FlyGeneric.kt (hunk 2 above)
Comment (Ell1ott) [reply src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/movement/fly/modes/FlyGeneric.kt]: Ah yes, that's not intended. I have corrected it

**Review by superblaubeere27 — APPROVED** [review]
👍🏻

**Comment by superblaubeere27** [comment]
Thanks again for the contribution! Please resolve the conflicts so we can merge it.

**Comment by Ell1ott** [comment]
> Thanks again for the contribution! Please resolve the conflicts so we can merge it.

Done 👍

**Comment by Ell1ott** [comment]
@superblaubeere27 Can this be merged?

**Pull request merged**